services/provider/server/server.go

```diff
@@ -422,6 +422,45 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 
 	}
 
+	cbpList := &rookCephv1.CephBlockPoolList{}
+	err = s.client.List(ctx, cbpList, client.InNamespace(s.namespace))
```

Comment:
List only the ones with the annotation set, you have no interest in any other ceph block pool